src/contracts/strategies/StrategyFactory.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+import "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import "./StrategyFactoryStorage.sol";
+import "./StrategyBase.sol";
+import "../permissions/Pausable.sol";
+
+/**
+ * @title Factory contract for deploying BeaconProxies of a Strategy contract implementation for arbitrary ERC20 tokens
+ *        and automatically adding them to the StrategyWhitelist in EigenLayer.
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ * @dev This may not be compatible with non-standard ERC20 tokens. Caution is warranted.
+ */
+contract StrategyFactory is StrategyFactoryStorage, OwnableUpgradeable, Pausable {
+    uint8 internal constant PAUSED_NEW_STRATEGIES = 0;
+
+    /// @notice EigenLayer's StrategyManager contract
+    IStrategyManager public immutable strategyManager;
+
+    /// @notice Since this contract is designed to be initializable, the constructor simply sets the immutable variables.
+    constructor(IStrategyManager _strategyManager) {
```

Comment:
We should have a way to prevent strategy deployments of LST tokens that already exist. Recommend adding this in initializer

Reply:
I had something like this before and ended up deleting it, after Scott pointed out that -- at least with the functions as I had laid them out -- admin actions would have made it possible to violate the "no duplicate strategies" property.

IMO the storage in this contract is currently primarily serving preventing duplicates as well as providing a nice little `token => strategy` connection, which we currently lack elsewhere (although I agree this excludes existing Strategies).

Existing strategies may vary (like the EIGEN) strategy and require a different upgrade process (due to being TUPs instead of BeaconProxies), so it's debatable whether they should be in the same mapping; we could always maintain two mappings though.

I've also thought about maintaining a _list_ of all deployed strategies from this contract, similar to how the Uniswap Factory (V2, at least) keeps a list of all deployed pairs, but we should always be able to index this list based on events, so it's of debatable value.

I think this is the area where we have the least precision on what we want + could make changes though! Perhaps @0xClandestine can lead figuring out the best path forwards here.